webapi/src/main/java/org/sakaiproject/webapi/controllers/SiteEntityController.java

```diff
@@ -0,0 +1,513 @@
+
+                    Optional.ofNullable(patchEntity.getOpenDate())
+                            .ifPresent(openDate -> assignment.setOpenDate(openDate));
+
+                    Optional.ofNullable(patchEntity.getDueDate())
+                            .ifPresent(dueDate -> assignment.setDueDate(dueDate));
+
+                    Optional.ofNullable(patchEntity.getCloseDate())
+                            .ifPresent(closeDate -> assignment.setCloseDate(closeDate));
+
+                    Optional.ofNullable(patchEntity.getGroupRefs())
+                            .ifPresent(groupRefs -> {
+                                Set<String> assignmentGroups = assignment.getGroups();
+
+                                // Remove old groups
+                                assignmentGroups.clear();
+
+                                if (groupRefs.isEmpty()) {
+                                    // Assign assignment to entire site
+                                    assignment.setTypeOfAccess(Assignment.Access.SITE);
+                                } else {
+                                    // Assign assignment to groups
+                                    assignment.setTypeOfAccess(Assignment.Access.GROUP);
+                                    // Add new groups
+                                    assignmentGroups.addAll(groupRefs);
+                                }
+                            });
+
+                    Assignment updatedAssignment;
+                    try {
+                        assignmentService.updateAssignment(assignment);
+                        updatedAssignment = assignmentService.getAssignment(assignment.getId());
+                    } catch (PermissionException | IdUnusedException e) {
+                        // Permission already evaluated before and assignmentId should be safe
+                        log.error("Pervious permission check was not sufficient: {}",
+                                ExceptionUtils.getStackTrace(e));
+                        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).build();
+                    }
+                    updatedEntities.add(SiteEntityRestBean.of(updatedAssignment));
+                    break;
+                case RESOURCE:
+                    ContentResourceEdit resourceEdit = (ContentResourceEdit) toolEntities.get(entityKey(patchEntity));
+                    Optional<Set<String>> optGroupRefs = Optional.ofNullable(patchEntity.getGroupRefs());
+                    Instant resourceOpenDate = patchEntity.getOpenDate();
+                    Instant resourceCloseDate = patchEntity.getCloseDate();
+                    boolean updateAvailability = ObjectUtils.allNotNull(resourceOpenDate, resourceCloseDate);
+
+                    if (updateAvailability) {
+                        resourceEdit.setAvailabilityInstant(false, resourceOpenDate, resourceCloseDate);
```

Comment:
TODO: This seems to not work right for some resource types, as the UI does not show the availability correctly